test/react-native-cli/features/steps/steps.rb

```diff
@@ -24,3 +24,97 @@
 When('I run the React Native service interactively') do
   step("I run the service '#{current_fixture}' interactively")
 end
+
+When('I press enter') do
+  step('I input "" interactively')
+end
+
+When('I wait for the shell to output the following to stdout') do |expected|
+  wait = Maze::Wait.new(timeout: MazeRunner.config.receive_requests_wait)
+
+  success = wait.until do
+    stdout = Runner.interactive_session.stdout_lines.join("\n")
+
+    stdout.include?(expected)
+  end
+
+  assert(
+    success,
+    <<~ERROR
+      Did not find the expected message in stdout
+
+      stdout:
+      #{Runner.interactive_session.stdout_lines.inspect}
+    ERROR
+  )
+end
+
+Then("I wait for the shell to output a line containing {string} to stdout") do |expected|
+  wait = Maze::Wait.new(timeout: MazeRunner.config.receive_requests_wait)
+  success = wait.until do
+    Runner.interactive_session.stdout_lines.any? do |line|
+      line.include?(expected)
+    end
+  end
+
+  assert(
+    success,
+    <<~ERROR
+      No stdout lines contained '#{expected}'
+
+      stdout:
+      #{Runner.interactive_session.stdout_lines.inspect}
+    ERROR
+  )
+end
+
+Then("I wait for the current stdout line to contain {string}") do |expected|
+  wait = Maze::Wait.new(timeout: MazeRunner.config.receive_requests_wait)
+  wait.until do
+    Runner.interactive_session.current_buffer.include?(expected)
+  end
+
+  assert_includes(Runner.interactive_session.current_buffer, expected)
+end
+
+def parse_package_json
+  before = Runner.interactive_session.stdout_lines.dup
+
+  steps %Q{
```

Comment:
Yeah I don't feel great about this, but I couldn't think of another way to do it. It would be nice to have the ability to use `docker-compose exec` in MR to make this a bit more robust

Reply:
Feel free to ticket that and we can look at working it in at some point.

Reply:
This comment sounded a bit more negative than I meant it to, sorry!

Created PLAT-5565 to add `docker-compose exec` support to MR